src/App.tsx

```diff
@@ -1,41 +1,92 @@
-import React from 'react';
-import './App.scss';
-import './styles/general.scss';
+import React, { useCallback, useEffect, useState } from 'react';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Loader } from './components/Loader';
+
+import { Post } from './types/Post';
+import { User } from './types/User';
+import { SelectedPostId } from './types/SelectedPostId';
+
+import { getAllPosts, getUserPosts } from './api/posts';
+import { getAllUsers } from './api/users';
+
+import './App.scss';
+import './styles/general.scss';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedPostId, setSelectedPostId] = useState<SelectedPostId>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const fetchPosts = useCallback(async () => {
+    setPosts(await getAllPosts());
```

Comment:
better not to do like this, create a separate variable

```suggestion
    const postsFromServer = await getAllPosts();

    setPosts(postsFromServer);
```

Reply:
done